tools/build

```diff
@@ -0,0 +1,128 @@
+#!/bin/sh
+
+set -ex
+
+print() {
+    printf '%s\n' "$*"
+}
+
+die() {
+    st=${?:-0}
+    if [ $st -eq 0 ]; then
+        st=2
+    fi
+    print "$*" >&2
+    exit $st
+}
+
+usage() {
+    die 'build [ -n -pg1[234] ] ( test-crates | test-extension | test-post-install | test-doc | clippy )'
+}
+
+require_pgversion() {
+    [ -n "$pgversion" ] || die 'specify one of -pg12 | -pg13 | -pg14'
+}
+
+find_pgconfig() {
+    require_pgversion
+    full_version=$(
+        cd "$HOME/.pgx"
+        set -- $pgversion.*/pgx-install
+        [ "$1" = "$pgversion.*/pgx-install" ] && die "$pgversion not installed in $HOME/.pgx"
+        set -- $(dirname $*)
+        [ $# -eq 1 ] || die "too many installations for $pgversion, found: $*"
+        echo $1)
+    pg_config="$HOME/.pgx/$full_version/pgx-install/bin/pg_config"
+}
+
+[ $# -ge 1 ] || usage
+
+# For reasons we don't yet understand, pgx prevents the cargo cache from
+# working across runs UNLESS all pgx commands are run from the 'extension'
+# subdirectory of the project.  Any cargo command we need to run from that
+# subdirectory, we use CARGO_TARGET_DIR="$extension_CARGO_TARGET_DIR", and for
+# all others, CARGO_TARGET_DIR="$top_CARGO_TARGET_DIR".
+#
+# If each entry in ci.yml `jobs` is guaranteed to be run in isolation, always
+# with the target dir freshly copied from the specified cache key, then we
+# don't need to bother with CARGO_TARGET_DIR here.
```

Comment:
I _think_ this holds. If I understand correctly, each job [will run in its own VM](https://docs.github.com/en/actions/learn-github-actions/understanding-github-actions#the-components-of-github-actions) which is isolated from the VMs running the other jobs. Don't know if that will simplify anything.

Reply:
I think so too, I just wanted to capture the conditions.